backend/routes/user.ts

```diff
@@ -273,21 +279,36 @@ export async function userModSelf(
             return ormLU
                 .getLoginUserById(checked.userId)
                 .then((user) => util.getOrReject(user))
-                .then((user) => {
-                    if (
-                        checked.data.pass != undefined &&
-                        user.password != checked.data.pass.oldpass
-                    )
-                        return Promise.reject();
+                .then(async (user) => {
+                    if (checked.data.pass != undefined) {
+                        let valid = true;
+                        if (checked.data.pass.oldpass) {
+                            valid = await bcrypt.compare(
+                                checked.data.pass.oldpass,
+                                user?.password || ""
+                            );
+                        }
+                        if (!valid) {
+                            return Promise.reject();
+                        }
+                    }
                     return Promise.resolve(user);
                 })
-                .then((user) =>
-                    ormLU.updateLoginUser({
-                        loginUserId: checked.userId,
-                        accountStatus: user.account_status,
-                        password: checked.data.pass?.newpass,
-                    })
-                )
+                .then(async (user) => {
+                    if (checked.data.pass?.newpass !== undefined) {
+                        const hash = await bcrypt.hash(
+                            checked.data.pass.newpass,
+                            config.encryption.encryptionRounds
+                        );
+                        return ormLU.updateLoginUser({
+                            loginUserId: checked.userId,
+                            accountStatus: user.account_status,
+                            password: hash,
+                        });
+                    } else {
+                        return Promise.resolve(user);
+                    }
+                })
                 .then((user) => Promise.resolve(user.person))
                 .then((person) => {
```

Comment:
Not really a fan of nested `.then()` but if the backend people find this easier then using `await` that's fine by me.